score/ScoreBonus.py

```diff
@@ -0,0 +1,330 @@
+from nameinternal import get_card_metadata, get_score_bonus
+from collections import Counter, namedtuple
+from save import Savefile
+from abc import ABC, abstractmethod
+
+# In general, these should only be needed for the Ongoing run
+# If for a completed run, use RunHistoryPlus's score_breakdown field
+class ScoreBonus(ABC):
+    def __init__(self, name: str):
+        data = get_score_bonus(name)
+        self.name = data["NAME"]
+        self.description = ""
+        if data["DESCRIPTIONS"]:
+            self.description = data["DESCRIPTIONS"][0]
+
+    def __init__(self, name: str, description: str) -> None:
+        self.name = name
+        self.description = description
+
+    @abstractmethod
+    def bonus_for_run(self, save: Savefile):
+        raise NotImplementedError
+
+    def get_starting_max_hp(self, save: Savefile):
+        max_hp_tuple = namedtuple("MaxHP", ["base_max_hp", "asc_max_hp_loss"])
+        max_hp_dict = {
+            "Ironclad": max_hp_tuple(80, 5),
+            "Silent": max_hp_tuple(70, 4),
+            "Defect": max_hp_tuple(75, 4),
+            "Watcher": max_hp_tuple(72, 4)
+        }
+
+        character_max_hp = max_hp_dict[save.character].base_max_hp
+        if save.ascension_level >= 14:
+            character_max_hp -= max_hp_dict[save.character].asc_max_hp_loss
+
+        return character_max_hp
+
+class Ascension(ScoreBonus):
+    def __init__(self):
+        super().__init__("Ascension")
+
+    def bonus_for_run(self, save: Savefile):
+        # Only applies to the score from the following bonuses:
+        # Floors Climbed
+        # Enemies Killed
+        # Elites Killed
+        # Bosses Slain
+        # Champion
+        # Perfect
+        # Beyond Perfect
+        # Overkill
+        # C-c-c-combo.
+        return 0
+
+class FloorsClimbed(ScoreBonus):
+    def __init__(self):
+        super().__init__("Floors Climbed", "Number of the Floor you reached")
+
+    def bonus_for_run(self, save: Savefile):
+        return 5 * save.current_floor
+
+class EnemiesKilled(ScoreBonus):
+    def __init__(self):
+        super().__init__("Enemies Killed", "For each Monster encounter defeated")
+
+    def bonus_for_run(self, save: Savefile):
+        return 2 * save._data.get("monsters_killed", 0)
```

Comment:
There should instead be a property on `FileParser` that accesses this. External code should not touch `_data`